vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -136,8 +136,10 @@ def parse_container_definition(container_def, pod_id)
         :cpu_cores         => container_def["cpu"].to_f / 1000
       }
       ports = container_def["ports"]
-      new_result[:container_port_configs] = ports.collect do |port_entry|
-        parse_container_port_config(port_entry, pod_id, container_def["name"])
+      unless ports.nil?
```

Comment:
Actually the nil check is more accurate.  .present? will also prevent blank arrays coming through.  But we *want* blank arrays because that will set the key and ultimately delete everything in the database.  If you don't set the key it will be ignored.